opentelemetry-api/src/opentelemetry/context/propagation/httptextformat.py

```diff
@@ -15,17 +15,19 @@
 import abc
 import typing
 
-from opentelemetry.trace import SpanContext
+from opentelemetry.context import Context
 
-_T = typing.TypeVar("_T")
+ContextT = typing.TypeVar("ContextT")
 
-Setter = typing.Callable[[_T, str, str], None]
-Getter = typing.Callable[[_T, str], typing.List[str]]
+Setter = typing.Callable[[ContextT, str, str], None]
+Getter = typing.Callable[[ContextT, str], typing.List[str]]
 
 
-class HTTPTextFormat(abc.ABC):
+class HTTPExtractor(abc.ABC):
```

Comment:
That's a good point. I'll rename the API from `HTTPExtractor` and `HTTPInjector` to `Extractor` and `Injector`, since it is generic enough at this point. I'll also move that interface into the `propagation/__init__.py` module directly. This can be changed later if it no longer makes sense, once the spec is flushed out for the propagation API.

Reply:
There are three things because of which "HTTP" kinda makes sense
1. To distinguish them from binary injectors (HTTP implies text based).
2. To distinguish them from injectors which just return a single string object (namely the binary ones) (HTTP injectors are based on key-value maps)
3. (Probably) to indicate the that the injector uses only HTTP-compatible keys and values (e.g. ASCII, starts with a-zA-Z, contains no spaces)

Maybe TextMapExtractor would be a better name. I think the 3rd point does not necessarily need to be in the interface. If we still want that, OpenTracing-cpp had an empty interface for HTTPHeader deriving from the corresponding TextMap interface (e.g. https://github.com/opentracing/opentracing-cpp/blob/4bb431f7728eaf383a07e86f9754a5b67575dab0/include/opentracing/propagation.h#L176), we could do the same so we could check `isinstance(myTextMapExtractor, HTTPExtractor)`.

Reply:
Good call @Oberon00, I missed that `Getter` and `Setter` specified `str` keys.

How would you expect the binary versions to work now? They're not specified in https://github.com/tedsuo/rfcs/blob/context-prop-2/text/0066-separate-context-propagation.md#Propagation-API, but the API changes to (HTTP) injectors/extractors would seem to make https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-propagators.md#binary-format obsolete.

What would it look like ff we split this up into http/text and binary versions? Something like this?

```python
TextSetter = typing.Callable[[CarrierT, str, str], None]
TextGetter = typing.Callable[[CarrierT, str], Sequence[str]]
```

```python
BinarySetter = typing.Callable[[CarrierT, T], None]
BinaryGetter = typing.Callable[[CarrierT], T]
```

(Note that I changed `ContextT` here to `CarrierT` to make it clear this is the carrier arg.)

Under the old spec, `T` would be a `SpanContext` or `DistributedContext`, and we set/get all fields in the carrier at once. Under the new spec, `T` could be any "concern".

It's not clear to me why we shouldn't be able to set/get individual keys for binary carriers as well as text carriers. Now that the carrier is generic, what would we lose by making the key and value types generic as well?